tests/mps_simulator_test.cc

```diff
@@ -237,11 +239,460 @@ TEST(MPSSimulator, Apply1InteriorArbitrary) {
   ASSERT_NEAR(state.get()[l_offset + 63], 153.6, 1e-5);
 }
 
+TEST(MPSSimulator, Apply2Left01) {
+  auto sim = MPSSimulator<For, float>(1);
+  using MPSStateSpace = MPSSimulator<For, float>::MPSStateSpace_;
+  auto ss = MPSStateSpace(1);
+
+  auto mps = ss.CreateMPS(3, 4);
+  memset(mps.get(), 0, ss.RawSize(mps));
+  mps.get()[0] = -0.35590581027809776;
+  mps.get()[1] = 0.026818375951314005;
+  mps.get()[2] = 0.8727655590119396;
+  mps.get()[3] = 0.3330047088601117;
+  mps.get()[8] = -0.6775160618958234;
+  mps.get()[9] = -0.6431047824209531;
+  mps.get()[10] = -0.1296437424681659;
+  mps.get()[11] = -0.33253671532297485;
+  mps.get()[16] = 0.5941347479820251;
+  mps.get()[17] = 0.19108928740024567;
+  mps.get()[18] = 0.585273027420044;
+  mps.get()[19] = -0.24035339057445526;
+  mps.get()[24] = 0.38375747203826904;
+  mps.get()[25] = 0.15926049649715424;
+  mps.get()[26] = -0.2938160300254822;
+  mps.get()[27] = -0.12502236664295197;
+  mps.get()[32] = -0.46030130982398987;
+  mps.get()[33] = -0.40587466955184937;
+  mps.get()[34] = 0.41750138998031616;
+  mps.get()[35] = -0.3110700249671936;
+  mps.get()[40] = 0.2459115833044052;
+  mps.get()[41] = -0.027767818421125412;
+  mps.get()[42] = -0.4402020573616028;
+  mps.get()[43] = -0.181321382522583;
+  mps.get()[80] = -0.40654852986335754;
+  mps.get()[81] = 0.0;
+  mps.get()[82] = 0.7775961756706238;
+  mps.get()[83] = 0.0723680704832077;
+  mps.get()[84] = 0.42058154940605164;
+  mps.get()[85] = 0.0;
+  mps.get()[86] = 0.21800334751605988;
+  mps.get()[87] = 0.02028878591954708;
+
+  std::vector<float> mat = {
+      1.0, 0.0, 0.0,  0.0, 0.0, 0.0, 0.0, 0.0, 0.0,   0.0,  0.0,
+      0.0, 0.0, -1.0, 0.0, 0.0, 0.0, 0.0, 0.0, -1.0,  0.0,  0.0,
+      0.0, 0.0, 0.0,  0.0, 0.0, 0.0, 0.0, 0.0, 0.866, -0.5,
+  };
+  sim.ApplyGate({0, 1}, mat.data(), mps);
+
+  float *wf = new float[32];
```

Comment:
Same comment as the previous PR - a clarifying comment on how these `wf` values are generated could be helpful.

For sections with Cirq equivalents, something like this might provide the best comparison: https://github.com/quantumlib/qsim/blob/cae957073623e343d0bc3b3627ab03082c59faef/tests/simulator_testfixture.h#L471
(includes an equivalent Cirq code snippet)

Reply:
Added a link to Cirq code that regenerates the fuzz test as well as diagrams and explanations for the other tests. Was this what you had in mind ?

Reply:
Yes, thank you!